optimizely/optimizely_config.py

```diff
@@ -209,24 +208,31 @@ def stringify_conditions(self, conditions, audiences_map):
             else:
                 return conditions[0].upper() + \
                     ' "' + self.lookup_name_from_id(conditions[1], audiences_map) + '"'
+        # If length is 2 (where the one elemnt is a list) or greater
         if length > 1:
             for i in range(length):
+                # Operand is handled here and made Upper Case
                 if conditions[i] in ARGS:
-                    condition = conditions[i].upper()
+                    operand = conditions[i].upper()
                 else:
+                    # Check if element is a list or not
                     if type(conditions[i]) == list:
+                        # Check if at the end or not to determine where to add the operand
+                        # Recursive call to call stringify on embedded list
                         if i + 1 < length:
                             conditions_str += '(' + self.stringify_conditions(conditions[i], audiences_map) + ') '
                         else:
-                            conditions_str += condition + \
+                            conditions_str += operand + \
                                 ' (' + self.stringify_conditions(conditions[i], audiences_map) + ')'
+                    # Not a list so we handle as and ID to lookup no recursion needed
```

Comment:
this comment is a bit difficult to understand?

Reply:
Updated to be more readable.